### PR TITLE
Improving active asset transitions

### LINKED ISSUE #630: Dashboard UX: fix graph loading reactivity issues
The first load animation for the Dashboard is weird for some reason, likely because the time range is set once on initial load, and again after everything loads.

How I reproduce:
- create a new metrics set
- hit quick start to save yoruself time
- go to the dashboard

This is probably due to the ordering of the different dispatches. Ideally we will not have to unravel a lot of stuff to make this work. Every other load seems to work great, so this feels like it's about instantiation.

### LINKED ISSUE #631: Dashboard UX: reference value doesn't properly update when I change time ranges
The reference value does not seem to update correctly when I switch between a short date range and “all time”. This messes with the bars used in the leaderboards.

### LINKED ISSUE #671: Dashboard UX : Updates to model does not update dashboard.
Updates to model query especially by editing the model file directly will not update the dashboard.

We need to add a robust update to the dashboard based on multiple changes,
1. Changing measure expression can sometimes not update the dashboard.
2. Changing the model query by adding where clause does not update the dashboard.
3. Messing with the sources tied to the model does not update the dasboard.

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #671 closes #630 closes #631

Fixing various asset sync issues. Moving various reactive statement to instead directly update on change in source.
1. Modifications to model updates dashboard.
2. Modifications to metrics definition updates dashboard.
3. Updating time range selection to update when various things change in metrics definition.

### CODE REVIEW

File: src/lib/redux-store/explore/bootstrapMetricsExplorer.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { createAsyncThunk } from "$lib/redux-store/redux-toolkit-wrapper";
+import { EntityType } from "$common/data-modeler-state-service/entity-state-service/EntityStateService";
+import { bootstrapMetricsDefinition } from "$lib/redux-store/metrics-definition/bootstrapMetricsDefinition";
+import type { RillReduxState } from "$lib/redux-store/store-root";
+import { selectMetricsExplorerById } from "$lib/redux-store/explore/explore-selectors";
+import { syncExplore } from "$lib/redux-store/explore/explore-apis";
+import { selectDimensionsByMetricsId } from "$lib/redux-store/dimension-definition/dimension-definition-selectors";
+import { selectMeasuresByMetricsId } from "$lib/redux-store/measure-definition/measure-definition-selectors";
+
+export const bootstrapMetricsExplorer = createAsyncThunk(
```
Comment (hamilton) [inline src/lib/redux-store/explore/bootstrapMetricsExplorer.ts:10]: what do we mean by bootstrap here? I am finding this language confusing when I see it. Does this mean that we're syncing everything needed to make the explore up-to-date?

**Review by hamilton — COMMENTED** [review]
Took a look at this PR ~ would prefer a second set of eyes since I've done too much context-switching today, but it looks good and seems to function as expected.

It does feel like we've basically re-implemented cache invalidation. After this upcoming moment has passed, I'd like us to spent a moment to intentionally plan on how we re-align the application state so we don't have to do so much in this particular way.

**Review by hamilton — APPROVED** [review]

File: src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte [hunk 2]
```diff
@@ -118,7 +118,7 @@
       <Tooltip location="right" alignment="center" distance={8}>
         <BarAndLabel
           showBackground={nullCount !== 0}
-          color={DATA_TYPE_COLORS[type].bgClass}
+          color={DATA_TYPE_COLORS[type]?.bgClass}
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte:121]: I got an unrecoverable error here once. Couldnt reproduce once the page loaded after i added this.

File: src/lib/redux-store/explore/bootstrapMetricsExplorer.ts (hunk 1 above)
Comment (AdityaHegde) [reply src/lib/redux-store/explore/bootstrapMetricsExplorer.ts:10]: It loads dimensions and measures if not already loaded. It also loads leaderboards, time series and big numbers. Hence the name, bootstrap everything related to the explorer.

File: src/lib/redux-store/big-number/big-number-apis.ts [hunk 3]
```diff
@@ -44,6 +52,49 @@ export const generateBigNumbersApi = createAsyncThunk(
...
+        thunkAPI.dispatch(setReferenceValues(id, referenceValues));
+      }
+    }
+
+    thunkAPI.dispatch(setBigNumber(id, bigNumbers));
+    thunkAPI.dispatch(setBigNumberStatus(id, EntityStatus.Idle));
+  }
+);
+
+const getBigNumberApi = createAsyncThunk(
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply src/lib/redux-store/big-number/big-number-apis.ts:85]: Adding a comment to mention the difference.

File: src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (hamilton) [reply src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte:121]: strange!

**Review by bcolloran — APPROVED** [review]
lgtm!

**Pull request merged**